### DISCUSSION TITLE
What's missing for 1.0?

### DISCUSSION THREAD

**Discussion by jmeistrich** [LegendApp/legend-state#72] [General]
As of version 0.22 we have finished everything we wanted to get into 1.0 and feel good about the API, so we are planning to release an RC version shortly and finish stabilizing towards 1.0. 

What do you think is missing for a 1.0 release?
- Are there any bugs or missing features that should be solved before 1.0?
- Are there any type issues that should be improved?
- Is anything not documented well enough or confusing?

**Comment by GiFarina** [comment]
I think that sample apps, documentation and virtualized list component like FlatList are missing

**Reply by jmeistrich** [reply]
In terms of samples, I recently made https://legendapp.com/open-source/state/react-examples/ with some larger examples. Would you like to see more examples like that? Or are you looking for more complete full app examples?

**Reply by GiFarina** [reply]
I was thinking of something more like [reanimated](https://github.com/software-mansion/react-native-reanimated/) example apps or something more friendly to read or another option is like guides/how to. I can help with these things. It would be necessary to define the road map

**Reply by jmeistrich** [reply]
Helping with that would be great! The current documentation is admittedly barebones and could use a lot of expanding on. And since I'm very close to it an outside perspective would help a lot.

In terms of roadmap, pending any incoming suggestions or issues to the contrary, I think the core and React integrations are pretty solid. So my upcoming plans are:
- **Remote persistence plugins** - I have a Firebase Realtime Database plugin used in Bravely and Legend which needs some cleanup before open sourcing. I'd like to also have Firestore and Supabase plugins, but I'm not experienced with them so those could benefit from others' help.
- **More helper observables** (like `currentTime` and `pageHash`) - I have no specific plans but I think there's a lot of potential, and there's probably many features of Bravely and Legend I could extract.
- **More hooks** - Our hooks would need to return observables and *not* trigger re-render. Also no specific plans, but lots of potential.
- **Other frameworks** - I'd love to see deep integrations with other frameworks, but I'm mainly experience with React so that could use some help too.

And of course it needs more/better documentation and example apps. At the moment I'm focusing on adding more tests, integrating it deeper into Legend, and hunting for bugs. So documenting and making examples is wide open if you or anyone else wants to jump in :)
